lib/rubocop/cop/rspec/no_let.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for usage of `let` blocks in specs.
+      #
+      # This cop can be configured with the option `AllowSubject` which
+      # will configure the cop to only register offenses on explicit calls to
+      # `let` and not calls to `subject`
+      #
+      # @example
+      #   # bad
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     it { expect(foo).to be_empty }
+      #   end
+      #
+      #   describe MyClass do
+      #     subject(:foo) { [] }
+      #     it { expect(foo).to be_empty }
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #     it do
+      #       foo = []
+      #       expect(foo).to be_empty
+      #     end
+      #   end
+      #
+      # @example with AllowSubject configuration
+      #
+      #   # rubocop.yml
+      #   # RSpec/NoLet:
+      #   #   AllowSubject: true
+      #
+      #   # bad
+      #   describe MyClass do
+      #     let(:foo) { [] }
+      #     it { expect(foo).to be_empty }
+      #   end
+      #
+      #   # good
+      #   describe MyClass do
+      #     subject(:foo) { [] }
+      #     it { expect(foo).to be_empty }
+      #   end
+      #
+      #   describe MyClass do
+      #     it do
+      #       foo = []
+      #       expect(foo).to be_empty
+      #     end
+      #   end
+      #
+      class NoLet < Cop
+        MSG = 'Avoid using `%<method>s` ' \
+              '– use a method call or local variable instead.'
+
+        def_node_matcher :let?, <<~PATTERN
+          (send nil? :let ...)
```

Comment:
This misses `let!`. Better utilize the language module - `Helpers::ALL.send_pattern`